From: Peter Kokot <peterkokot@gmail.com>
Subject: Add CMake changes for PHP-8.4 branch

* CMake added to run-tests.php for info about failing tests and checks
---
 run-tests.php | 24 ++++++++++++++++++------
 1 file changed, 18 insertions(+), 6 deletions(-)

diff --git a/run-tests.php b/run-tests.php
index 4654ead998..a36dbf0f24 100755
--- a/run-tests.php
+++ b/run-tests.php
@@ -865,10 +865,12 @@ function write_information(array $user_tests, $phpdbg): void
         <?php
         $exts = get_loaded_extensions();
         $ext_dir = ini_get('extension_dir');
-        foreach (scandir($ext_dir) as $file) {
-            if (preg_match('/^(?:php_)?([_a-zA-Z0-9]+)\.(?:so|dll)$/', $file, $matches)) {
-                if (!extension_loaded($matches[1]) && @dl($matches[1])) {
-                    $exts[] = $matches[1];
+        if (is_dir($ext_dir)) {
+            foreach (scandir($ext_dir) as $file) {
+                if (preg_match('/^(?:php_)?([_a-zA-Z0-9]+)\.(?:so|dll)$/', $file, $matches)) {
+                    if (!extension_loaded($matches[1]) && @dl($matches[1])) {
+                        $exts[] = $matches[1];
+                    }
                 }
             }
         }
@@ -957,7 +959,7 @@ function save_results(string $output_file, bool $prompt_to_save_results): void
 
     $failed_tests_data .= "\n" . $sep . 'BUILD ENVIRONMENT' . $sep;
     $failed_tests_data .= "OS:\n" . PHP_OS . " - " . php_uname() . "\n\n";
-    $ldd = $autoconf = $sys_libtool = $libtool = $compiler = 'N/A';
+    $ldd = $autoconf = $sys_libtool = $libtool = $compiler = $cmake = 'N/A';
 
     if (!IS_WINDOWS) {
         /* If PHP_AUTOCONF is set, use it; otherwise, use 'autoconf'. */
@@ -968,7 +970,9 @@ function save_results(string $output_file, bool $prompt_to_save_results): void
         }
 
         /* Always use the generated libtool - Mac OSX uses 'glibtool' */
-        $libtool = shell_exec(INIT_DIR . '/libtool --version');
+        if (file_exists(INIT_DIR . '/libtool')) {
+            $libtool = shell_exec(INIT_DIR . '/libtool --version');
+        }
 
         /* Use shtool to find out if there is glibtool present (MacOSX) */
         $sys_libtool_path = shell_exec(__DIR__ . '/build/shtool path glibtool libtool');
@@ -990,9 +994,17 @@ function save_results(string $output_file, bool $prompt_to_save_results): void
         }
 
         $ldd = shell_exec("ldd $php 2>/dev/null");
+
+        /** If PHP_CMAKE is set, use it; otherwise, use 'cmake'. */
+        if (getenv('PHP_CMAKE')) {
+            $cmake = shell_exec(getenv('PHP_CMAKE') . ' --version');
+        } else {
+            $cmake = shell_exec('cmake --version');
+        }
     }
 
     $failed_tests_data .= "Autoconf:\n$autoconf\n";
+    $failed_tests_data .= "CMake:\n$cmake\n";
     $failed_tests_data .= "Bundled Libtool:\n$libtool\n";
     $failed_tests_data .= "System Libtool:\n$sys_libtool\n";
     $failed_tests_data .= "Compiler:\n$compiler\n";
